src/coreclr/vm/i386/jithelp.S

```diff
@@ -552,55 +552,16 @@ LOCAL_LABEL(LRszMORE32):
 LEAF_END JIT_LRsz, _TEXT
 
 // *********************************************************************/
-//  JIT_Dbl2LngP4x87
-//
-// Purpose:
-//   converts a double to a long truncating toward zero (C semantics)
-//
-// uses stdcall calling conventions
-//
-//   This code is faster on a P4 than the Dbl2Lng code above, but is
-//   slower on a PIII.  Hence we choose this code when on a P4 or above.
-//
-LEAF_ENTRY JIT_Dbl2LngP4x87, _TEXT
-    // get some local space
-    sub 	esp, 8
-
-    #define arg1 [esp + 0x0C]
-    fld     QWORD PTR arg1          // fetch arg
-    fnstcw  WORD PTR arg1           // store FPCW
-    movzx   eax, WORD PTR arg1      // zero extend - wide
-    or      ah, 0x0C                // turn on OE and DE flags
-    mov     DWORD PTR [esp], eax    // store new FPCW bits
-    fldcw   WORD PTR  [esp]         // reload FPCW with new bits
-    fistp   QWORD PTR [esp]         // convert
-
-    // reload FP result
-    mov	    eax, DWORD PTR [esp]
-    mov	    edx, DWORD PTR [esp + 4]
-
-    // reload original FPCW value
-    fldcw   WORD PTR arg1
-    #undef arg1
-
-    // restore stack
-    add     esp, 8
-
-    ret
-LEAF_END JIT_Dbl2LngP4x87, _TEXT
-
-// *********************************************************************/
-//  JIT_Dbl2LngSSE3
+//  JIT_Dbl2LngFpu
 //
 // Purpose:
 //    converts a double to a long truncating toward zero (C semantics)
 //
 //  uses stdcall calling conventions
 //
-//    This code is faster than the above P4 x87 code for Intel processors
-//    equal or later than Core2 and Atom that have SSE3 support
+//    This code is faster than C++ even on modern CPUs but it requires SSE3
 //
-LEAF_ENTRY JIT_Dbl2LngSSE3, _TEXT
```

Comment:
What is `Fpu` suffix trying to say?

I think `JIT_Dbl2LngSSE3` was a better name for this method.

Reply:
Wanted to note it as using FPU instructions which are generally discouraged these days. I honestly don't know which one is the better name here.

Reply:
I think that the SSE3 is a better name. The distinct characteristic of this implementation is that it requires SSE3. It is the best implementation possible assuming SSE3 as a baseline. I do not think that the details that it still uses x87 floating point instructions should be captured in the name.

(Consider hypothetical case where we would have multiple implementations of this helper for multiple different x86 baselines - AVX, AVX2, ... . Including the baseline ID in the name would be the most sensible naming scheme.)

Reply:
Also, touching this helper is a throw away change. As I have said before, my preference would be to omit all throw-away changes from this PR.